src/averageAges.js

```diff
@@ -33,8 +47,19 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(person => person.sex === 'f');
+
+  if (withChildren) {
+    const womenWithChildren = people.filter(woman => {
+      return people.some(child => child.mother === woman.name);
+    });
+
+    return countAvaregeAge(womenWithChildren);
+  } else {
+    return countAvaregeAge(women);
+  }
```

Comment:
```suggestion
    const womenWithChildren = people.filter(woman => {
      return people.some(child => child.mother === woman.name);
    });
  return countAvaregeAge(
    withChildren
      ? womenWithChildren
      : women
    )
```
Тернарником можно красивее сделать

Reply:
Согласен, выглядит намного круче